src/DynamicRupture/Output/ReceiverBasedOutput.cpp

```diff
@@ -0,0 +1,398 @@
+                                          std::array<real, 6>& rotatedStress) {
+
+  auto& impAndEta = ((local.layer->var(drDescr->impAndEta))[local.ltsId]);
+  local.slipRateStrike = -impAndEta.invEtaS * (rotatedUpdatedStress[3] - rotatedStress[3]);
+  local.slipRateDip = -impAndEta.invEtaS * (rotatedUpdatedStress[5] - rotatedStress[5]);
+}
+
+void ReceiverBasedOutput::computeSlipRate(const double* tangent1,
+                                          const double* tangent2,
+                                          const double* strike,
+                                          const double* dip) {
+  local.slipRateStrike = static_cast<real>(0.0);
+  local.slipRateDip = static_cast<real>(0.0);
+
+  for (size_t i = 0; i < 3; ++i) {
+    real factorMinus = (local.faceAlignedValuesMinus[7] * tangent1[i] +
+                        local.faceAlignedValuesMinus[8] * tangent2[i]);
+
+    real factorPlus = (local.faceAlignedValuesPlus[7] * tangent1[i] +
+                       local.faceAlignedValuesPlus[8] * tangent2[i]);
+
+    local.slipRateStrike += (factorMinus - factorPlus) * strike[i];
+    local.slipRateDip += (factorMinus - factorPlus) * dip[i];
+  }
+}
+
+real ReceiverBasedOutput::computeRuptureVelocity(Eigen::Matrix<real, 2, 2>& jacobiT2d) {
+  auto* ruptureTime = (local.layer->var(drDescr->ruptureTime))[local.ltsId];
+  real ruptureVelocity = 0.0;
+
+  bool needsUpdate{true};
+  for (size_t point = 0; point < misc::numberOfBoundaryGaussPoints; ++point) {
+    if (ruptureTime[point] == 0.0) {
+      needsUpdate = false;
+    }
+  }
+
+  if (needsUpdate) {
+    constexpr int numPoly = CONVERGENCE_ORDER - 1;
+    constexpr int numDegFr2d = (numPoly + 1) * (numPoly + 2) / 2;
+    std::array<double, numDegFr2d> projectedRT{};
+    projectedRT.fill(0.0);
+
+    std::array<double, 2 * numDegFr2d> phiAtPoint{};
+    phiAtPoint.fill(0.0);
+
+    auto chiTau2dPoints =
+        init::quadpoints::view::create(const_cast<real*>(init::quadpoints::Values));
+    auto weights = init::quadweights::view::create(const_cast<real*>(init::quadweights::Values));
+
+    auto* rt = local.layer->var(drDescr->ruptureTime);
+    for (size_t jBndGP = 0; jBndGP < misc::numberOfBoundaryGaussPoints; ++jBndGP) {
+      real chi = chiTau2dPoints(jBndGP, 0);
+      real tau = chiTau2dPoints(jBndGP, 1);
+      basisFunction::TriDubiner::evaluatePolynomials(phiAtPoint.data(), chi, tau, numPoly);
+
+      for (size_t d = 0; d < numDegFr2d; ++d) {
+        projectedRT[d] += weights(jBndGP) * rt[local.ltsId][jBndGP] * phiAtPoint[d];
+      }
+    }
+
+    auto m2inv =
+        seissol::init::M2inv::view::create(const_cast<real*>(seissol::init::M2inv::Values));
+    for (size_t d = 0; d < numDegFr2d; ++d) {
+      projectedRT[d] *= m2inv(d, d);
+    }
+
+    real chi = chiTau2dPoints(local.nearestInternalGpIndex, 0);
+    real tau = chiTau2dPoints(local.nearestInternalGpIndex, 1);
+
+    basisFunction::TriDubiner::evaluateGradPolynomials(phiAtPoint.data(), chi, tau, numPoly);
```

Comment:
Probably, people didn't want to make a space interpolation of a solution.

Reply:
I agree Lukas that would make much more sense.
I suppose I use the nearest GP approach, moved towards the center of the point, to avoid inaccurate results when the cell center is outside the area most sampled by the gauss points.

Reply:
I think it will require a dedicated PR